util/hatchet/_generate_hatchet_token.sh

```
@@ -0,0 +1,59 @@
+PROJECT_ROOT="$(cd "$SCRIPT_DIR/../.." && pwd)"
+
+# Use docker compose (v2) if available, otherwise fall back to docker-compose
+if docker compose version &> /dev/null; then
+    COMPOSE_CMD="docker compose"
+else
+    COMPOSE_CMD="docker-compose"
+fi
+
+# Check if required Hatchet services are running (from project root)
+echo "🔍 Checking if Hatchet services are running..."
+cd "$PROJECT_ROOT"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)